### PR TITLE
Consolidate parsing of RFC 3339 datetime strings.

### LINKED ISSUE #1146: Consolidate parsing of RFC 3339 datetime strings
See: https://github.com/GoogleCloudPlatform/gcloud-python/pull/1145/files#r40020506

`gcloud._helpers._RFC3339_MICROS` is used to parse datetime strings in:
- `gcloud.dns.changes` (after #1145 lands)
- `gcloud.pubsub.message`
- `gcloud.storage.bucket`
- `gcloud.storage.blob` (twice)

### PR DESCRIPTION
**Pull request by dhermes**
Adding helpers for conversion both to and from strings. Fixes #1146.

### CODE REVIEW

File: gcloud/_helpers.py [hunk 1]
```diff
@@ -298,6 +298,31 @@ def _total_seconds(offset):
         return offset.total_seconds()
 
 
+def _str_to_datetime(dt_str):
```
Comment (tseaver) [inline gcloud/_helpers.py]: Hmm, seems like a too-vague name:  Maybe `_rfc3339_to_datetime`?

File: gcloud/_helpers.py [hunk 2]
```diff
@@ -298,6 +298,31 @@ def _total_seconds(offset):
...
+    :param dt_str: The string to convert.
+
+    :rtype: :class:`datetime.datetime`
+    :returns: The datetime object created from the string.
+    """
+    return datetime.datetime.strptime(
+        dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+
+
+def _datetime_to_str(value):
```
Comment (tseaver) [inline gcloud/_helpers.py]: Likewise, maybe `_datetime_to_rfc3339`?

File: gcloud/_helpers.py (hunk 1 above)
Comment (dhermes) [inline gcloud/_helpers.py]: SGTM. Will ping once updated.

**Comment by dhermes** [comment]
@tseaver PTAL. Replaced with

```
git grep -l _str_to_datetime | xargs sed -i s/_str_to_datetime/_rfc3339_to_datetime/g
git grep -l _datetime_to_str | xargs sed -i s/_datetime_to_str/_datetime_to_rfc3339/g
```

**Comment by tseaver** [comment]
LGTM

**Pull request merged**